routing/router.go

```diff
@@ -2700,147 +2701,40 @@ func (r *ChannelRouter) BuildRoute(amt *lnwire.MilliSatoshi,
 	hops []route.Vertex, outgoingChan *uint64,
 	finalCltvDelta int32, payAddr *[32]byte) (*route.Route, error) {
 
-	log.Tracef("BuildRoute called: hopsCount=%v, amt=%v",
-		len(hops), amt)
-
-	var outgoingChans map[uint64]struct{}
-	if outgoingChan != nil {
-		outgoingChans = map[uint64]struct{}{
-			*outgoingChan: {},
-		}
-	}
-
-	// If no amount is specified, we need to build a route for the minimum
-	// amount that this route can carry.
-	useMinAmt := amt == nil
-
-	// We'll attempt to obtain a set of bandwidth hints that helps us select
-	// the best outgoing channel to use in case no outgoing channel is set.
-	bandwidthHints, err := newBandwidthManager(
-		r.cachedGraph, r.selfNode.PubKeyBytes, r.cfg.GetLink,
-	)
-	if err != nil {
-		return nil, err
+	// TODO: Allow min amount route building via path finding.
```

Comment:
The required fix is to let pathfinding increase the running amount to match `min_htlc`. Useful independently too.

Reply:
#6927 explores the idea